security/pkg/server/ca/monitoring.go

```diff
@@ -67,8 +66,7 @@ var (
 	)
 	certChainExpiryTimestamp = monitoring.NewGauge(
 		"citadel_server_cert_chain_expiry_timestamp",
-		"The unix timestamp, in seconds, when Citadel cert chain will expire. "+
-			"A negative time indicates the cert is expired.",
+		"The unix timestamp, in seconds, when Citadel cert chain will expire.",
```

Comment:
If I understand the code correctly, it checks the key bundle - which may also be loaded from cacerts. 

I don't know if the metric is set only for the self-signed cert ( due the convoluted code path) - or not, but I think the correct comment should reflect what the code does ( i.e. if we set this metric for both cacerts and self-signed or not).

The 'citadel' comment is EXTREMELY confusing - citadel is the old internal name for a deployment/component that implemented certificate provisioning and has 5 ways to generate certs, including using K8S delegation. 

Maybe s/Citadel/Istio generated certificates/ ?

Reply:
And most important - please confirm if this is set in the different modes ( self-signed with Secret, intermediary certs from file or secret, RA to K8S,  mounted Istiod certificates) - should be based on the use of key bundle, but you can never be sure with the spaghetti we have.

Reply:
@costinm Does this changing the name of the gauge too?

Reply:
@costinm I'm going to keep the names the same on this PR unless you feel very strongly.
I tried three different cert modes.
With self signed certs (not doing anything), then only the `citadel_server_root_*` metrics will show.
If I [plug in my generated certs](https://istio.io/latest/docs/tasks/security/cert-management/plugin-ca-cert/) I get both `citadel_server_root_*` and `citadel_server_cert_chain_*` metrics.
Finally, if I use `cert-manager`, then I don't see anything.

I installed everything using the default profile and I used these instructions for cert-manager https://cert-manager.io/docs/usage/istio-csr/installation/

Reply:
Yes, I agree, keep the names. We can add new names - in new PRs, perhaps without citadel in them - but existing metrics are part of the API surface and should be treated as such.

I meant to update the documentation/comment - not the metric name.